pandas/_libs/tslibs/conversion.pyx

```diff
@@ -294,9 +296,15 @@ cdef _TSObject convert_to_tsobject(object ts, tzinfo tz, str unit,
         if ts == NPY_NAT:
             obj.value = NPY_NAT
         else:
-            ts = cast_from_unit(ts, unit)
+            if unit is None:
+                in_reso = abbrev_to_npy_unit("ns")
```

Comment:
should we assign unit="ns" here since we pass it to cast_from_unit below?

Reply:
OK for follow-up

Reply:
yup, good one, thanks